File: orchestrator/web/src/types/workload.ts
export interface WorkloadPackage {
  packageId: string
  version: string
  preInitSteps?: string[]
  postInitSteps?: string[]
}

export interface Workload {
  id: number
  workloadId: string
  workloadName: string
  version: string
  uploadedAt: string
  packages: WorkloadPackage[]
}

File: orchestrator/backend/Models/Workload.cs
namespace Orchestrator.Models;

public class Workload
{
    public int Id { get; set; }
    public string WorkloadId { get; set; } = string.Empty;
    public string WorkloadName { get; set; } = string.Empty;
    public string Version { get; set; } = string.Empty;
    public string DefinitionPath { get; set; } = string.Empty;
    public DateTime UploadedAt { get; set; } = DateTime.UtcNow;
    public ICollection<WorkloadPackage> Packages { get; set; } = [];
}
